From 8826614ff6200d0b759f48f86c505887bc4e71ac Mon Sep 17 00:00:00 2001
From: Craig Kewley <craigx.kewley@intel.com>
Date: Fri, 18 Aug 2017 09:33:50 +0100
Subject: [PATCH] Revert "ASoC: Codecs: TDF8532: remove codec"

This reverts commit 306d63eff42073d767aec05c594239e7ab7de3ea.

Conflicts:
	sound/soc/codecs/Kconfig
	sound/soc/codecs/Makefile
---
 sound/soc/codecs/Kconfig   | 5 +++++
 sound/soc/codecs/Makefile  | 2 ++
 sound/soc/codecs/tdf8532.c | 6 ++++++
 3 files changed, 13 insertions(+)

diff --git a/sound/soc/codecs/Kconfig b/sound/soc/codecs/Kconfig
index a05a58d..d9093fa 100644
--- a/sound/soc/codecs/Kconfig
+++ b/sound/soc/codecs/Kconfig
@@ -74,6 +74,7 @@ config SND_SOC_ALL_CODECS
 	select SND_SOC_DIO2125
 	select SND_SOC_DMIC if GPIOLIB
 	select SND_SOC_ES8316 if I2C
+	select SND_SOC_TDF8532 if I2C
 	select SND_SOC_ES8328_SPI if SPI_MASTER
 	select SND_SOC_ES8328_I2C if I2C
 	select SND_SOC_ES7134
@@ -557,6 +558,10 @@ config SND_SOC_DIO2125
 config SND_SOC_DMIC
 	tristate
 
+config SND_SOC_TDF8532
+    tristate "NXP Semiconductors TDF8532 Codec"
+	depends on I2C
+
 config SND_SOC_HDMI_CODEC
 	tristate
 	select SND_PCM_ELD
diff --git a/sound/soc/codecs/Makefile b/sound/soc/codecs/Makefile
index 8a6ee2c..fedef12 100644
--- a/sound/soc/codecs/Makefile
+++ b/sound/soc/codecs/Makefile
@@ -71,6 +71,7 @@ snd-soc-da9055-objs := da9055.o
 snd-soc-dmic-objs := dmic.o
 snd-soc-es7134-objs := es7134.o
 snd-soc-es8316-objs := es8316.o
+snd-soc-tdf8532-objs := tdf8532.o
 snd-soc-es8328-objs := es8328.o
 snd-soc-es8328-i2c-objs := es8328-i2c.o
 snd-soc-es8328-spi-objs := es8328-spi.o
@@ -316,6 +317,7 @@ obj-$(CONFIG_SND_SOC_DA9055)	+= snd-soc-da9055.o
 obj-$(CONFIG_SND_SOC_DMIC)	+= snd-soc-dmic.o
 obj-$(CONFIG_SND_SOC_ES7134)	+= snd-soc-es7134.o
 obj-$(CONFIG_SND_SOC_ES8316)    += snd-soc-es8316.o
+obj-$(CONFIG_SND_SOC_TDF8532) += snd-soc-tdf8532.o
 obj-$(CONFIG_SND_SOC_ES8328)	+= snd-soc-es8328.o
 obj-$(CONFIG_SND_SOC_ES8328_I2C)+= snd-soc-es8328-i2c.o
 obj-$(CONFIG_SND_SOC_ES8328_SPI)+= snd-soc-es8328-spi.o
diff --git a/sound/soc/codecs/tdf8532.c b/sound/soc/codecs/tdf8532.c
index c2bf5d8..04ab478 100644
--- a/sound/soc/codecs/tdf8532.c
+++ b/sound/soc/codecs/tdf8532.c
@@ -91,6 +91,12 @@ static int tdf8532_dai_trigger(struct snd_pcm_substream *substream, int cmd,
 		/* WA on unexpected codec down during S3
 		 SNDRV_PCM_TRIGGER_STOP fails so skip set ret */
 		tdf8532_stop_play(tdf8532);
+		data[4] = 0x27;
+		ret = tdf8532_cmd_send(tdf8532, data, ARRAY_SIZE(data));
+
+		/*delay 300ms to allow state change to occur*/
+		/*TODO: add state check to wait for state change*/
+		mdelay(300);
 		break;
 	}
 
-- 
1.9.1

